statistics/histogram.go

```diff
@@ -1105,6 +1128,29 @@ func (idx *Index) outOfRange(val types.Datum) bool {
 	return !withInLowBoundOrPrefixMatch || !withInHighBound
 }
 
+// RemoveUpperBound removes the upper bound the index stats.
+// It is used for when merge stats for incremental analyze.
+func (idx *Index) RemoveUpperBound(sc *stmtctx.StatementContext, values []types.Datum) (*Histogram, *CMSketch, error) {
+	hist, cms := idx.Histogram.Copy(), idx.CMSketch.copy()
+	buckets := hist.Buckets
+	buckets[hist.Len()-1].Count -= hist.Buckets[hist.Len()-1].Repeat
+	hist.Buckets[hist.Len()-1].Repeat = 0
```

Comment:
It is safe to set it to 0 because:

- If the histogram of the incremental analyze contains the upper bound, then when merging them, the `repeat` will be set to the real `repeat`.
- If it does not contains it, it is correct anyway.

Reply:
But for the second case, `ContainsFeedback` may return true in this case?

Reply:
Oh, I mean the histogram of the incremental one, not the current one.